marshall/file_read.js

```diff
@@ -0,0 +1,111 @@
+const fs = require('fs');
+const EventEmitter = require('events').EventEmitter;
+const ee = new EventEmitter();
+
+ee.on('file-read', (data) => {
+  console.log('READ: ', data.toString());
+})
+
+
+// Could this function be an example of nested callbacks?
+
+function makeReadFile(txt){
+  return function(){
+    fs.readFile(txt, (err, data) => {
+      console.log(data);
+      ee.emit('file-read', data);
+    });
+  };
+};
+
+var one = makeReadFile('./one.txt');
+var two = makeReadFile('./two.txt');
+var three = makeReadFile('./three.txt');
+
+one();
+two();
+three();
+
+///----------------- OR OR OR (is this worse?)
+
+ee.on('one', () => {
+  process.nextTick(() => {
+    fs.readFile('./one.txt', (err, data) => {
+      console.log(data);
+      ee.emit('file-read', data);
+    });
+  });
+});
+
+ee.on('two', () => {
+  process.nextTick(() => {
+    fs.readFile('./two.txt', (err, data) => {
+      console.log(data);
+      ee.emit('file-read', data);
+    });
+  });
+});
+
+ee.on('three', () => {
+  process.nextTick(() => {
+    fs.readFile('./three.txt', (err, data) => {
+      console.log(data);
+      ee.emit('file-read', data);
+    });
+  });
+});
+
+ee.emit('one');
+ee.emit('two');
+ee.emit('three');
+
```

Comment:
It seems like you are trying to use a couple different things together here in a way that doesn't quite accomplish the task that you are setting out to accomplish.  It looks like you are grasping the way the event emitters work.  But, you are just not implementing them as well as you could be.